### ISSUE TITLE
🐛 [Bug]: agent.Struct fails to unmarshal response since 2.33.0

### ISSUE THREAD

**Issue by teemukataja** [gofiber/fiber#2134]
### Bug Description

- `agent.Struct()` raises a null pointer exception.
- `agent.Struct()` works in `2.32.0`, but stopped working after `2.33.0`, including the latest release `2.38.1`.

#### Working version with 2.32.0
1. set fiber version to `2.32.0` in `go.mod`
2. run `go mod tidy`
3. run script
4. response is successfully parsed

#### Broken version 2.33.0-2.38.1
1. set fiber version to anything after `2.33.0` in `go.mod`
2. run `go mod tidy`
3. run script
4. null pointer exception is raised

### Expected Behavior

JSON response is unmarshalled into struct.

### Fiber Version

2.33.0-2.38.1

### Code Snippet (optional)

```go
package main

import (
	"fmt"

	"github.com/gofiber/fiber/v2"
)

type response struct {
	Title string `json:"title,omitempty"`
}

func main() {

	agent := fiber.AcquireAgent()
	defer agent.ConnectionClose()
	request := agent.Request()
	request.Header.SetMethod("GET")
	request.SetRequestURI("https://jsonplaceholder.typicode.com/todos/1")

	_ = agent.Parse()

	var r response
	_, _, _ = agent.Struct(&r)  // exception raised here

	fmt.Println(r)
}
```
```
$ go run cmd/main.go
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x668e38]

goroutine 1 [running]:
github.com/gofiber/fiber/v2.(*Agent).Struct(0xc0000fa000, {0x680720, 0xc000013180})
	/home/go/1.18.6/pkg/mod/github.com/gofiber/fiber/v2@v2.38.1/client.go:829 +0xf8
main.main()
	/home/go/1.18.6/src/request-test/cmd/main.go:24 +0x1c5

```

### Checklist:

- [X] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/master/.github/CODE_OF_CONDUCT.md).
- [X] I have checked for existing issues that describe my problem prior to opening this one.
- [X] I understand that improperly formatted bug reports may be closed without explanation.

**Comment by li-jin-gou** [comment]
use `a.JSONDecoder(json.Unmarshal).Struct(&r)`

**Comment by ReneWerner87** [comment]
@teemukataja thanks for the hint, fixed it in the pull request #2137 

will be in the next release

until then you have to set the default decoder by hand
https://github.com/gofiber/fiber/issues/2134#issuecomment-1265283424
